test/unit/CMakeLists.txt

```diff
@@ -1,5 +1,16 @@
 # CMake configuration for PROJ unit tests
 
+option(USE_EXTERNAL_GTEST "Compile against external GTest" OFF)
+
+if (USE_EXTERNAL_GTEST)
+
+message(STATUS "Using external GTest")
+find_package(GTest 1.8.0)
```

Comment:
How is this supposed to work? The Gtest configuration file exports prefixed targets, e.g. `GTest::gtest`. But the targets below link unprefixed targets, e.g. `gtest`. Thus the build rules do not pick up the include directories carried by the target etc.
So IMO the build rules need to use the prefix, and for consistency, a `find_package` with the fixed internal path should be added to the `else` branch (internal GTest).

BTW it wouldn't hurt to declare this `REQUIRED`.

BTW 1.8.0 will not work with MinGW. 1.8.1 is fixed, https://github.com/google/googletest/pull/721.

BTW later in this file there is a test commented as "ph_phi2_test not compatible of a .dll build". The surrounding tests needs to use WIN32 instead of MSVC, because its not compatible with MinGW, too.

I can prepare a PR with these changes if desired:
- [ ] find_package and prefix corrections
- [ ] googletest 1.8.1
- [ ] ph_phi2_test  correction

Reply:
> I can prepare a PR with these changes if desired

Please do. But be aware that I am preparing a release candidate tomorrow so you need to work fast.

Reply:
Have you tried it? I just tried prefixing them and it doesn't understand it (with GTest 1.8.1.)

Reply:
> Have you tried it? I just tried prefixing them and it doesn't understand it (with GTest 1.8.1.)

Yes, I tried with `USE_EXTERNAL_GTEST=ON` and CMake 3.11 or so.

Did you check the command line how the tests are compiled? I couldn't spot the include directories before the change. I guess it works for you because you have GTest in the compiler's standard directories. But these are not always the right ones for cmake, depending on `CMAKE_FIND_...` configuration.

Reply:
No, I mean CMake (3.12.1) doesn't configure. It's true that everything here is in a standard directory though, for the working build.

If you open a PR, I can try out what you mean.

Reply:
Created #1265.